mlrun/api/crud/projects.py

```diff
@@ -176,9 +177,30 @@ def delete_project_resources(
 
         # delete project secrets - passing None will delete all secrets
         if mlrun.mlconf.is_api_running_on_k8s():
-            mlrun.api.utils.singletons.k8s.get_k8s_helper().delete_project_secrets(
-                name, None
+            secrets = None
+            (
+                secret_name,
+                _,
+            ) = mlrun.api.utils.singletons.k8s.get_k8s_helper().delete_project_secrets(
+                name, secrets
             )
+            if mlrun.mlconf.events.mode == mlrun.common.schemas.EventsMode.disabled:
+                return
+            try:
+                client = (
+                    mlrun.api.utils.events.events_factory.EventsFactory().get_events_client()
+                )
+                event = client.generate_project_secrets_deleted_event(
+                    name, secret_name, secrets
+                )
+                client.emit(event)
+            except Exception as exc:
+                logger.warning(
+                    "Failed to emit project secrets deleted event",
+                    exc=exc,
+                    project=name,
+                    secret_names=secrets,
+                )
```

Comment:
that entire flow is irrelevant to the crud itself and makes the project flow a bit dirty. Id go with 
1. nop client unless stated otherwise, and by that remove the `if mlrun.mlconf.events.mode == mlrun.common.schemas.EventsMode.disabled:`
2. have `emit` not throwing anything (wrap it entirely under try/except)

evetually, the flow would be

```python
client = mlrun.api.utils.events.events_factory.EventsFactory().get_events_client()
client.emit(client. compile_project_secrets_deleted_event())
```